### PR TITLE
docs: add available configuration options for rollouts controller. Fixes #1914

### LINKED ISSUE #1914: JSON Logging
# Summary

Argo Rollouts is one of the few Argo projects that doesn't support JSON logging

# Use Cases

When trying to naively parse logs using a log aggregator like Fluentd.

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritize the issues with the most 👍.

### PR DESCRIPTION
**Pull request by hcelaloner**
It seems that docs are not listing available configuration options for the controller, this is a just simple attempt to list options as in ArgoCD Server Configuration Parameters docs.

Refs: #1914
Signed-off-by: Celal Öner <celal.oner@trendyol.com>

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-rollouts/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [x] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [x] I've signed my commits with [DCO](https://github.com/argoproj/argoproj)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My builds are green. Try syncing with master if they are not. 
* [ ] My organization is added to [USERS.md](https://github.com/argoproj/argo-rollouts/blob/master/USERS.md).

### CODE REVIEW

File: docs/installation.md [hunk 1]
```diff
@@ -26,6 +26,73 @@ This will create a new namespace, `argo-rollouts`, where Argo Rollouts controlle
...
 
+### Configuration Options
+
+Argo Rollouts Controller is a Kubernetes controller that continuously manages the Rollout resources in the cluster. This command runs Argo Rollouts Controller in the foreground, and it can be configured by following options.
+
+```
+rollouts-controller [flags]
+```
+
+```
```
[the remark this answers is not in the document]
Comment (hcelaloner) [reply docs/installation.md:37]: Yes, you are right. On the negative side, when a new option is added/removed, this section could contain stale information in the future. However, on the other hand, someone can see all available configuration options from docs. If this seems unnecessary I can completely remove this part and only gave an example of changing log format.

By the way, how about writing something like as follows after the example as an alternative 

You can list all available cli options for rollouts controller using the following command:
```shell
docker run --rm -it quay.io/argoproj/argo-rollouts:v1.2.0 -h
```

[the remark this answers is not in the document]
Comment (huikang) [reply docs/installation.md:37]: The alternative looks much better. Please go ahead with the proposed change. Thanks.

**Review by huikang — APPROVED** [review]

File: docs/installation.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (hcelaloner) [reply docs/installation.md]: First, thanks for the comments and review 🙂. I changed the docs as we talked but some checks are failing. I guess it is not related to the changes proposed in this pr, or is there actually something I should fix?

File: docs/installation.md [hunk 2]
```diff
@@ -26,6 +26,23 @@ This will create a new namespace, `argo-rollouts`, where Argo Rollouts controlle
 You can find released container images of the controller at [Quay.io](https://quay.io/repository/argoproj/argo-rollouts?tab=tags). There are also old releases
 at Dockerhub, but since the introduction of rate limiting, the Argo project has moved to Quay.
 
+### Configuration Options
+
+For instance, logging format can be easily set as `json` via passing the following additional command-line argument in Argo Rollouts Controller Deployment:
```
Comment (agrawroh) [inline docs/installation.md:31]:
```suggestion
For instance, logging format can be easily set as `json` via passing the following additional command-line argument in ArgoRollouts controller deployment:
```

File: docs/installation.md [hunk 3]
```diff
@@ -26,6 +26,23 @@ This will create a new namespace, `argo-rollouts`, where Argo Rollouts controlle
...
+
+````yaml
+      containers:
+      - image: "quay.io/argoproj/argo-rollouts:v1.2.0"
+        args:
+        - --logformat=json
+````
+
+You can list all available CLI options for rollouts controller using the following command:
```
Comment (agrawroh) [inline docs/installation.md:40]:
```suggestion
You can list all the available CLI options for the rollouts controller using the following command:
```

**Comment by hcelaloner** [comment]
Hi everyone, I tried to update the branch and apply all the pr suggestions. Could you review it again when you have time and merge it if it is okay for the project?

**Comment by hcelaloner** [comment]
Thanks to everyone who helped in the past. I'm closing this pr, it seems that there will be no activity for this pr.

**Pull request closed without merging**